industrial_ci/src/tests/source_tests.sh

```diff
@@ -79,11 +97,13 @@ function run_clang_tidy_check {
     local clang_tidy_args=()
     ici_parse_env_array clang_tidy_args CLANG_TIDY_ARGS
 
-    ici_run "install_clang_tidy" ici_install_pkgs_for_command clang-tidy clang-tidy "$(apt-cache depends --recurse --important clang  | grep "^libclang-common-.*")"
+    ici_run "install_clang_tidy" ici_install_pkgs_for_command clang-tidy \
+        clang-tidy "$(apt-cache depends --recurse --important clang  | grep "^libclang-common-.*")" \
+        "$(test -n "$CLANG_TIDY_BASE_REF" && echo git)"
```

Comment:
To install `git` please use `ici_setup_git_client`.

Reply:
One thing I just noticed.  Should we add an option to specify the version of clang-tidy?  There is an option to specify the version of clang-format.  I'm not sure if it matters as much with clang-tidy as I'm not sure if there are similar changes.  

Also, if we allow specifying the version how do we deal with someone wanting to specify a new version of clang-tidy on an old version of ROS where that version isn't available for install?  That being said, maybe ignore this comment for now.

Reply:
> Should we add an option to specify the version of clang-tidy? There is an option to specify the version of clang-format.

This is up to you. If you need it, just add it.

> Also, if we allow specifying the version how do we deal with someone wanting to specify a new version of clang-tidy on an old version of ROS where that version isn't available for install?

This is just not supported. Similar to clang-format.

Reply:
For our use-case the default versions of clang-tidy that are installed work the same for our config.  So this is a non-issue I think.  Maybe this is much less of an issue than with different versions of clang-format.

Reply:
> For our use-case the default versions of clang-tidy that are installed work the same for our config. So this is a non-issue I think. Maybe this is much less of an issue than with different versions of clang-format.

Sorry to derail this comment on this line, I should have just used a normal comment for this.

Reply:
Sure. Done.